charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -32,8 +32,15 @@ metrics:
     - pull:
         exporter:
           prometheus:
-            host: {{ ._0 | replace "{env!" "{env:" }}
-            port: {{ ._1 }}
+            host: {{ .address._0 | replace "{env!" "{env:" }}
+            port: {{ .address._1 }}
+            {{- if .Values.config.service.telemetry.resource }}
+            with_resource_constant_labels:
+              included:
+              {{- range (keys .Values.config.service.telemetry.resource) }}
+              - {{ println . }}
+              {{- end }}
```

Comment:
This handles a breaking change that happened with the otel sdk's prometheus exporter where resource attributes were not included by default